cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c

```diff
@@ -389,6 +389,28 @@ unsigned int test_ternary02(unsigned int x) {
   return y1 + y2 + y3 + y4 + y5;
 }
 
+// Test that nested ternary expressions of literals doesn't cause performance blow up.
+double test_ternary_nested_of_literals(double m, double n, double o, double p, double q) {
```

Comment:
Do we have another way to test against performance regressions? This test adds around 2 minutes of execution time on my machine prior to the PR, and runs with "normal" performance after. So the idea is that any regression would be noticed in the test time.

Reply:
This test is perfectly fine. By default there's a 5 min timeout limit on `codeql test run` so you could make the test take ~5 min prior to your fix (which would write something like "timeout" in the expected file) which would surely make it clear that there's a regression if we ever broke this again.

You can rewrite this into a macro and nest it a couple of times to make the test arbitrarily enormous :)

We normally don't add regression tests for performance issues. Ideally, we would add a DCA project that exhibit these performance problems, but this was only seen internally at Microsoft this is a bit more tricky.

Reply:
I've now added two more lines of nested ternary operators, which should grow the old execution time by a factor of 36 and get us well above the timeout.

In order to reproduce the blow up it's important that the constants are all very different and doesn't repeat. Otherwise the sums of the bounds can end up producing duplicates, and the blow up gets reduced. Given that I'm not sure about how to use a macro, so I've left the repetition be :)